File: leetcode-go/Q001-Q100/Q035-Search-Insert-Position/Q035-Search-Insert-Position.go
package Q035_Search_Insert_Position

func searchInsert(nums []int, target int) int {
	if nums[0] >= target {
		return 0
	}

	i := 0
	for ; i < len(nums); i++ {
		if nums[i] >= target {
			break
		}
	}
	return i
}

File: leetcode-java/src/linkedin/Q035_Search_Insert_Position.java
package linkedin;

public class Q035_Search_Insert_Position {

    public int searchInsert(int[] nums, int target) {
        int l = 0, r = nums.length;
        while (l <= r) {
            int m = l + (r-l)/2;
            if (nums[m] == target) {
                return m;
            }
            if (nums[m] < target) {
                l = m+1;
            } else {
                r = m-1;
            }
        }
        return l;
    }

}
